autosklearn/automl.py

```diff
@@ -1104,6 +1124,23 @@ def _create_search_space(self, tmp_dir, backend, datamanager,
     def configuration_space_created_hook(self, datamanager, configuration_space):
         return configuration_space
 
+    def __getstate__(self) -> Dict[str, Any]:
+        # Cannot serialize a client!
+        self._dask_client = None
+        return self.__dict__
+
+    def __del__(self):
+
+        # Make sure the client closes, so we don't leave
+        # the tcp connection open
+        if self._dask_client:
```

Comment:
This would also close a dask client if the user passed it as an argument. Could you please record whether that's the case and only close it if it was created by Auto-sklearn?

Reply:
I have created a new attribute as suggested.